src/1_utils.js

```diff
@@ -61,6 +61,42 @@ utils.userPreferences = {
 	}
 };
 
+utils.generateDynamicBNCLink = function(branchKey, data) {
+	if (!branchKey && !data) {
+		return;
+	}
+	if (!data["data"] || typeof data["data"] !== "string") {
+		data = utils.cleanLinkData(data);
+	}
+	var addKeyAndValueToUrl = function(fallbackUrl, tagName, tagData) {
+		var first = fallbackUrl[fallbackUrl.length - 1] === "?";
+		var modifiedFallbackURL = first ? fallbackUrl + tagName : fallbackUrl + "&" + tagName;
+		modifiedFallbackURL += "=";
+		return tagName === "data" ? modifiedFallbackURL + encodeURIComponent(utils.base64encode(tagData)) : modifiedFallbackURL + encodeURIComponent(tagData);
```

Comment:
I would recommend passing in the base64encoded value into the function instead of handling here in order to simplify the function.